lib/get-package-path.js

```diff
@@ -1,16 +1,17 @@
 const path = require('path');
 const loadJsonFile = require('load-json-file');
-const pkgDir = require('pkg-dir');
+const pkgUp = require('pkg-up');
 
 const DEFAULT_PACKAGE = {xo: false};
 
 // (base: string) => pkgPath: Promise<string>
 async function findPackage(base) {
-	const pkgPath = path.join(base, 'package.json');
+	const pkgPath = await pkgUp({cwd: base});
 	const pkg = await loadPkg(pkgPath);
 
-	if (pkg.xo === false && base !== null) {
-		return findPackage(path.join(base, '..'));
+	if (pkg.xo === false && pkgPath !== null) {
+		const newBase = path.resolve(path.dirname(pkgPath), '..');
+		return getPackagePath(newBase);
```

Comment:
Calculates a new `base` path from the directory above any `package.json` file found.

This way, `pkg-up` will handle _most_ of the directory traversal. If it finds a `package.json` file with `pkg.xo === false` it continues traversing up.

When `/` is reached, it returns `null`.